cloud/linode/loadbalancers.go

```diff
@@ -769,11 +766,16 @@ func getConnectionThrottle(service *v1.Service) int {
 	return connThrottle
 }
 
-func makeLoadBalancerStatus(nb *linodego.NodeBalancer) *v1.LoadBalancerStatus {
+func makeLoadBalancerStatus(service *v1.Service, nb *linodego.NodeBalancer) *v1.LoadBalancerStatus {
+	ingress := v1.LoadBalancerIngress{
+		Hostname: *nb.Hostname,
+	}
+	if !getServiceBoolAnnotation(service, annLinodeHostnameOnlyIngress) {
+		ingress.IP = *nb.IPv4
+	}
```

Comment:
It seems strange that the annotation for a NB hostname on the svc is opt-in, but the logic in here implies the other way around, that the default is to return the NB's hostname unless a specific annotation is set. Do you think it could maybe it read like:
```suggestion
	ingress := v1.LoadBalancerIngress{
		Hostname: *nb.IPv4,
	}
	if getServiceBoolAnnotation(service, annLinodeHostnameOnlyIngress) {
		ingress.IP = *nb.Hostname
	}
```
... or am I missing a subtle bug by doing this?

Reply:
Before the kube-proxy workaround, we returned both the IP and hostname. Now If the new annotation is present we only want the hostname. Otherwise, we want the old default of having the IP and hostname